enhancements/network-observability.md

```diff
@@ -0,0 +1,367 @@
+1. CNI must be OVN-Kubernetes<br>
+    The network type (CNI) has to be OVN-Kubernetes since configuring OVS to
+export IPFIX data is only supported there.  By default, the CNI is OpenShift
+SDN and the overwhelming majority of customers run OpenShift SDN today.
+Therefore in a brown field scenario, this would require the user to change
+their CNI type before network observability can be enabled.
+    Note that the effort to implement OpenShift SDN to configure OVS is
+doable.  However, new features recently added have only been implemented
+in OVN-Kubernetes.
+
+2. Data sampling<br>
+    In order to scale in terms of collecting data, it is not uncommon to take
+a very small sample of data. By default, the data sample rate in OVS is
+400 to 1.  Networks with larger bandwidth, sample at an even higher rate
+such as 1,024 to 1.
+
+    With sampling, you will be able to get trends about your network.  If
+you are looking for a specific flow of data, it's likely not to be there.
+The total number of bytes and packets are just estimations when sampling
+is used.  Therefore if you need full visibility of the traffic and accuracy,
+the only choice is not do sampling.
+
+3. Resources required<br>
+    Exporting data is potentially CPU, memory and bandwidth-intensive, and
+typically requires a large amount of storage.  The CPU/memory/bandwidth
+resources can be managed by the sampling data rate.  The storage requirements
+can be controlled by the data retention policy or how long to keep the data.
+
+    Nevertheless, in order to maintain the same performance that you have
+prior to enabling network observability, will require additional resources.
+
+
+### Risks and Mitigations
+
+1. Resource Risk<br>
+    Not having enough resources (e.g. CPU) can be disruptive to the system,
+and care must be taken to not jeopardize the health of the network by enabling
+network observability.
+
+2. Privacy<br>
+    Operators need to be aware of what data is being collected as it might
+expose customers to privacy issues.  In sensitive cases, storing IP addresses
+is problematic as it can reveal the user's general location and can be used
+to find out other information about the user, including possibly the user's
+identity.
+
+3. Cost<br>
+    It may not be clear how much additional cost the customer will incur,
+particularly if this is managed by Red Hat.  Additional nodes may need to
+be added to avoid the resource risk described above.
+
+4. Data accuracy<br>
+    If data sampling is turned on, then data accuracy cannot be relied on
+so using this for audit reports or criminal investigation are not viable.
+
+
+## Design Details
+
+### Software Architecture
+This is the software architecture at a high level.
+
+![architectural diagram](arch22.drawio.svg)
+
+The components with hatch marks indicate work that needs to be done.
+
+For the first release, it intentionally leaves out the message broker,
+the plan being Kafka, for simplicity.  For the specific choice of
```

Comment:
This is a pretty significant dependency that's only briefly mentioned and probably serves more discussion.  You could call out scalability as a risk and the consideration of kafka to scale out the feature as a way to mitigate that.  We'll also need to guide customers on what scale we expect the current work to be suitable for.